ext/curl/interface.c

```diff
@@ -2876,13 +2885,10 @@ PHP_FUNCTION(curl_setopt)
 
 	ch = Z_CURL_P(zid);
 
-	if (options <= 0 && options != CURLOPT_SAFE_UPLOAD) {
-		php_error_docref(NULL, E_WARNING, "Invalid curl configuration option");
-		RETURN_FALSE;
-	}
-
-	if (_php_curl_setopt(ch, options, zvalue) == SUCCESS) {
+	if (_php_curl_setopt(ch, options, zvalue, 0) == SUCCESS) {
 		RETURN_TRUE;
+	} else if (EG(exception)) {
+		RETURN_THROWS();
```

Comment:
I don't think this makes sense. Just leave the RETURN_FALSE.

Reply:
I've just get rid of these exception checks.